src/main/java/com/assignment/application/test/CompanyServiceTest.java

```diff
@@ -0,0 +1,172 @@
+        //result
+        Assert.assertEquals(companyTest,null);
+
+    }
+
+    @Test
+    public void testGetCompanyList() throws Exception {
+        //assumption
+        List<Company> companyList = new ArrayList<>();
+        companyList.add(new Company("Microsoft", "Technology", 1000000L, "California", "Bill Gates"));
+        companyList.add(new Company("Google", "Technology", 1000000L, "California", "Sergey Brin, Larry Page"));
+        //action
+        when(companyService.getCompanyList()).thenReturn(companyList);
+        //result
+        Assert.assertEquals(companyList.size(), 2);
+        Assert.assertEquals(companyList.get(0).getName(), "Microsoft");
+        Assert.assertEquals(companyList.get(0).getHeadOffice(), "California");
+        Assert.assertEquals(companyList.get(0).getIndustryType(), "Technology");
+        Assert.assertEquals(companyList.get(0).getFounder(), "Bill Gates");
+    }
+
+    @Test
+    public void testGetCompleteCompanyInfo(){
+        //assumption company exists
+        List<CompleteCompInfo> completeCompInfoList = new ArrayList<>();
+        completeCompInfoList.add(new CompleteCompInfo("Sundar Pichai","google_3",1L,"Engineering",2L,(double)1000000,"123454323454",null,"1234567890","California","Chennai"));
+        //action
+        when(companyService.getCompleteCompInfo(anyLong())).thenReturn(completeCompInfoList);
+        //result
+        Assert.assertEquals(completeCompInfoList.size(),1);
+        Assert.assertEquals(completeCompInfoList.get(0).getEmployeeName(),"Sundar Pichai");
+        Assert.assertEquals(completeCompInfoList.get(0).getEmployeeId(),"google_3");
+        Assert.assertEquals(Optional.ofNullable(completeCompInfoList.get(0).getCompanyId()), Optional.of(1L));
+        Assert.assertEquals(completeCompInfoList.get(0).getDepartmentName(),"Engineering");
+        Assert.assertEquals(Optional.ofNullable(completeCompInfoList.get(0).getDepartmentId()),Optional.of(2L));
+        Assert.assertEquals(Optional.ofNullable(completeCompInfoList.get(0).getSalary()),Optional.of(1000000.0));
+        Assert.assertEquals(completeCompInfoList.get(0).getAccountNo(),"123454323454");
+        Assert.assertEquals(completeCompInfoList.get(0).getProjectId(),null);
+        Assert.assertEquals(completeCompInfoList.get(0).getPhoneNumber(),"1234567890");
+        Assert.assertEquals(completeCompInfoList.get(0).getCurrentAdd(),"California");
+        Assert.assertEquals(completeCompInfoList.get(0).getPermanentAdd(),"Chennai");
+
+        //assumption if no such company exist
+        List<CompleteCompInfo> completeCompInfosEmpty = null;
+        //action
+        when(companyService.getCompleteCompInfo(anyLong())).thenReturn(completeCompInfosEmpty);
+        //result
+        Assert.assertEquals(completeCompInfosEmpty,null);
+    }
+
+    @Test
+    public void testUpdateCompanyInfo(){
+        //assumption company exists and company update json is not null
```

Comment:
Incorrect way of testing. You have put the method under test in when clause. Then, in essence, no behaviour is getting tested here.

Reply:
Applies to all tests written here. Please redo them all

Reply:
A typical test could be to check against null behaviour from repo. Or say, deleting or updating something that does not exist, or say finding something that exists as successful behaviour for success cases. There can be a lot many tests added.